kafka-connect-ftp/src/main/scala/com/datamountaineer/streamreactor/connect/ftp/source/FtpMonitor.scala

```diff
@@ -87,13 +118,58 @@ class FtpMonitor(settings:FtpMonitorSettings, fileConverter: FileConverter) exte
     }
   }
 
+  def bySlicesFetch(mode: MonitorMode, file: AbsoluteFtpFile, prevFetch: Option[FileMetaData]): Option[(Option[FileMetaData], FetchedFile)] = {
+    logger.info(s"fetching ${file.path}")
+
+    val offsetToReadFrom = prevFetch match {
+      case None =>
+        logger.info(s"First time file is read")
+        0
+      case Some(previous) if previous.offset == -1 =>
+        logger.info(s"First time file is read")
+        0
+      case Some(previous) if mode == MonitorMode.Update && (previous.attribs.timestamp != file.timestamp || previous.attribs.size != file.size) =>
+        logger.info(s"We are in Update mode && File was updated, restart reading from offset 0")
+        0
+      case Some(previous) =>
+        logger.info(s"Continue reading from offset ${previous.offset}")
+        previous.offset
+    }
+    logger.info(s"offset to read from = ${offsetToReadFrom}")
+
+    var buffer = new Array[Byte](sliceSize)
+
+    ftp.setRestartOffset(offsetToReadFrom)
+    val inputStream = ftp.retrieveFileStream(file.path)
+    logger.info(s"inputStream=${inputStream}")
+    val bytesRead = inputStream.read(buffer)
+    logger.info(s"bytesRead=${bytesRead}")
+    inputStream.close()
+    logger.info(s"inputStream.close()")
+    ftp.completePendingCommand()
+    logger.info(s"completePendingCommand()")
+
+    //Trim end
+    val bytes = util.Arrays.copyOfRange(buffer, 0, bytesRead)
+    // it doesn't make sense to compute the hash for chunks/slices
+    val hash = DigestUtils.sha256Hex(bytes)
+
+    val attributes = new FileAttributes(file.path, file.ftpFile.getSize, file.ftpFile.getTimestamp.toInstant)
+
+    val meta = prevFetch match {
+      case None => FileMetaData(attributes, hash, Instant.now, Instant.now, Instant.now)
```

Comment:
Those three `Instant.now` will be slightly different. Would it be more coherent to only call that method once ?

Reply:
Functionnaly it's not very important, landoop does the same. That data isn't used in any logic.